src/riscv-hybrid-integration.adoc

```diff
@@ -279,9 +279,20 @@ shown in xref:default-csrnames-added[xrefstyle=short].
 
 {cheri_default_ext_name} optionally allows <<MODESW>> to execute in debug mode.
 
-When entering debug mode, the core always enters {cheri_cap_mode_name}.
+When entering debug mode, whether the core enters {cheri_int_mode_name} or
```

Comment:
which CRE bit are you thinking of though?

I was thinking that a specific core will specify that it will always enter a specific mode on debug mode entry, and this will be matched against the debugger (integer mode if using a legacy debugger). 

Then the core can have an input pin driven by a fuse which controls which mode it is in when entering debug mode, which can be set to cap mode when it is known that a CHERI-aware debugger is in use.

That way legacy debuggers and CHERI aware debuggers can work, and the mode can be selectable by a pin or some other mechanism. the piont is that we don't specify any of that and leave it to the core vendor.

Alternatively we need to specify a control which says which mode to enter, but which one? Are you thinking of menvcfg.CRE? Or add another bit? That is another solution.

Reply:
I was thinking of using `mseccfg.CRE` as the bit that determines which mode the debugger enters into. The other option could be to use dinfc.mode and have that be intmode on reset - then we could allow the debugger to write an infinite cap with a different mode to specify the mode on entry. A bit ugly allowing a write there but it avoids allocating yet another register.

Reply:
this causes a problem though - you might have a core running CHERI code and need to connect it to a non-CHERI aware debugger - the non-CHERI aware debugger will be of limited use, but will be able to probe memory

the point of making the mode UNSPECIFIED is to allow legacy debuggers to be used even if the core is running CHERI code, as an essential step on the way to CHERI adoption.

so I think this PR is correct as it stands, and we shouldn't be enforcing extra restrictions on debugger usage

Reply:
That's not a use case to care much about. If you think of CHERI as a new base ISA it just doesn't really make sense to do.

Reply:
And there will be plenty of time to teach OpenOCD about capability mode prior to ratification.

Reply:
I really don't want to see the CHERI debugging experience made worse, nor put unnecessary burden on implementations (e.g. needing a larger program buffer to be supported so mode switching can be done), in the name of compatibility.

Reply:
This is a real use case that we have right now, and others will have it to when they move to CHERI cores.
My proposal of giving people the choice doesn't slow down or prevent openOCD being updated.
The recommendation is to be in capablity mode once debugger support is available.

Reply:
spec update, will request re-review